### PR TITLE
Added serialize function to Model hawkes

### PR DESCRIPTION
**Pull request by andro2157**
Added serialize function to Model Hawkes

Serialization and comparaison for hawkes models

### CODE REVIEW

**Comment by Dekken** [comment]
Added serialization and comparisons for survival module

**Comment by Dekken** [comment]
3.7 is failing due to time constrants, asked @MaryanMorel to reduce sccs test

File: tick/survival/tests/model_coxreg_partial_lik_test.py [hunk 1]
```diff
@@ -24,6 +24,22 @@ def test_ModelCoxRegPartialLik(self):
         model_spars.fit(csr_matrix(features), times, censoring)
         self.run_test_for_glm(model, model_spars)
 
+    def test_sccs_serialize_and_compare(self):
```
Comment (MaryanMorel) [inline tick/survival/tests/model_coxreg_partial_lik_test.py]: you meant `test_coxreg_serialize_and_compare`?

File: tick/survival/tests/model_coxreg_partial_lik_test.py [hunk 2]
```diff
@@ -24,6 +24,22 @@ def test_ModelCoxRegPartialLik(self):
...
+        np.random.seed(123)
+        n_samples, n_features = 100, 5
+        w0 = np.random.randn(n_features)
+        features, times, censoring = SimuCoxReg(
+            w0, n_samples=n_samples, verbose=False, seed=1234).simulate()
+        model = ModelCoxRegPartialLik()
+        model.fit(features, times, censoring)
+        model_spars = ModelCoxRegPartialLik()
+        model_spars.fit(csr_matrix(features), times, censoring)
+
```
Comment (MaryanMorel) [inline tick/survival/tests/model_coxreg_partial_lik_test.py]: Why are you creating a sparse version if you do not use it in the test assertions?

File: tick/survival/tests/model_sccs_test.py [hunk 3]
```diff
@@ -135,6 +135,24 @@ def _test_grad(self, model, coeffs, delta_check_grad=1e-5,
         self.assertAlmostEqual(
             norm(model.grad(coeffs_min)), .0, delta=delta_model_grad)
 
+    def test_sccs_serialize_and_compare(self):
+        """Test longitudinal multinomial model Lipschitz constant."""
```
Comment (MaryanMorel) [inline tick/survival/tests/model_sccs_test.py]: Docstring does not match with the test

File: tick/survival/tests/model_coxreg_partial_lik_test.py [hunk 4]
```diff
@@ -24,6 +24,22 @@ def test_ModelCoxRegPartialLik(self):
         model_spars.fit(csr_matrix(features), times, censoring)
         self.run_test_for_glm(model, model_spars)
 
+    def test_sccs_serialize_and_compare(self):
+        """Test longitudinal multinomial model Lipschitz constant."""
```
Comment (MaryanMorel) [inline tick/survival/tests/model_coxreg_partial_lik_test.py]: Docstring does not match with the test

File: lib/include/tick/survival/model_sccs.h [hunk 5]
```diff
@@ -96,6 +97,61 @@ class DLL_PUBLIC ModelSCCS : public ModelLipschitz {
       out[i] = exp(x[i] - x_max) / sum;  // overflow-proof
     }
   }
+
+  template <class Archive>
+  void serialize(Archive &ar) {
```
Comment (MaryanMorel) [inline lib/include/tick/survival/model_sccs.h]: Why this method as you are already implementing load and save?

Comment (Dekken) [reply lib/include/tick/survival/model_sccs.h]: typo ^_^

File: tick/survival/tests/model_coxreg_partial_lik_test.py (hunk 1 above)
Comment (Dekken) [reply tick/survival/tests/model_coxreg_partial_lik_test.py]: copy/pasta

File: tick/survival/tests/model_coxreg_partial_lik_test.py (hunk 2 above)
Comment (Dekken) [reply tick/survival/tests/model_coxreg_partial_lik_test.py]: copy/pasta again, I just took the top test and changed it (not enough!)

**Comment by Dekken** [comment]
solve shttps://github.com/X-DataInitiative/tick/issues/79

**Pull request merged**